### PR TITLE
feat: `inherit` field for release template inheritance

### PR DESCRIPTION
**Pull request by mumoshu**
Ref https://github.com/helmfile/helmfile/issues/435#issuecomment-1364749414

Signed-off-by: Yusuke Kuoka <ykuoka@gmail.com>

### CODE REVIEW

**Comment by yxxhero** [comment]
@mumoshu dos needs to be updated.

File: pkg/state/state_exec_tmpl.go [hunk 1]
```diff
@@ -132,3 +144,76 @@ func (st *HelmState) ExecuteTemplates() (*HelmState, error) {
...
+
+type CyclicReleaseTemplateInheritanceError struct {
+	Message string
+}
+
+func (e CyclicReleaseTemplateInheritanceError) Error() string {
+	return e.Message
+}
+
+func (st *HelmState) releaseWithInheritedTemplate(r *ReleaseSpec, inheritancePath []string) (*ReleaseSpec, error) {
```
Comment (yxxhero) [inline pkg/state/state_exec_tmpl.go:156]: @mumoshu could you add more comments for this function?

Comment (mumoshu) [reply pkg/state/state_exec_tmpl.go:161]: Added in 1d99e63 👍

**Comment by mumoshu** [comment]
@yxxhero I've updated the documentation via 5dcf252!

**Comment by yxxhero** [comment]
@mumoshu first implementation. LGTM.

**Review by yxxhero — APPROVED** [review]

**Comment by mumoshu** [comment]
@yxxhero Thanks for reviewing!

**Pull request merged**